spec/design_topics/data_interchange.md

```diff
@@ -58,18 +58,66 @@ means the object it is attached to must return a `numpy.ndarray`
 containing the data the object holds).
 ```
 
-TODO: design an appropriate Python API for DLPACK (`to_dlpack` followed by `from_dlpack` is a little clunky, we'd like it to work more like the buffer protocol does on CPU, with a single constructor function).
 
-TODO: specify the expected behaviour with copy/view/move/shared-memory semantics in detail.
+## Syntax for data interchange with DLPack
 
+The array API will offer the following syntax for data interchange:
 
-```{note}
+1. A `from_dlpack(x)` function, which accepts (array) objects with a
+   `__dlpack__` method and uses that method to construct a new array
+   containing the data from `x`.
+2. A `__dlpack__(self, stream=None)` method on the array object, which
+   will be called from within `from_dlpack`.
+
+
+## Semantics
+
+DLPack describe the memory layout of strided, n-dimensional arrays.
+When a user calls `y = from_dlpack(x)`, the library implementing `x` (the
+"producer") will provide access to the data from `x` to the library
+containing `from_dlpack` (the "consumer"). If possible, this must be
+zero-copy (i.e. `y` will be a _view_ on `x`). If not possible, that library
+may make a copy of the data. In both cases:
```

Comment:
It might be useful to also note somewhere that when the strides field is nullptr, it indicate a row-major compact array

Reply:
Yeah, and it'd be also nice to note if the array is of zero size, in the DLTensor its data pointer should set to nullptr (or 0). In the events where strides are not specified, this is the only effective way to inform the consumer that a zero-size array is coming. Numba/CuPy received a few bug reports when CAI v0/v1 were not clear about this, thus leading to the v2 revision.

Reply:
The two comments above do not seem to be addressed yet.

Reply:
Good catch, I missed these - thanks. Both points are added to the Implementation section now.